explainer/index.bs

```diff
@@ -23,19 +23,59 @@ Issue(tabatkins/bikeshed#2006): Set up cross-linking into the WebGPU and WGSL sp
-is needed (here, in the spec Introduction section, or both).
+## Use cases ## {#use-cases}
 
+Example use cases for WebGPU that aren't addressed by WebGL 2 are:
+
+ - Drawing images with highly-detailed scenes with many different objects (such as CAD models). WebGPU's drawing commands are individually cheaper than WebGL's.
+ - Execute advanced algorithms for drawing realistic scenes. Many modern rendering techniques and optimizations cannot execute on WebGL 2 due to the lack of support for general computations.
+ - Execute machine learning models efficiently on the GPU. WebGL can be massaged to execute machine learning but it is difficult and sub-optimal.
+ - Provide a foundation to expose modern GPU functionality on the Web. WebGPU exposes functionality that became available in native around 2010 but is structured similarly to all current native GPU APIs. There are plans to later extend it to have more modern functionality. See also: [[#why-not-webgl3]].
+
+Concrete examples are:
+
+ - Improving existing Javascript 3D libraries like Babylon.js and Three.js with new rendering techniques (compute-based particles, fancier post-processing, ...) and offloading on the GPU expensive computations currently done one the CPU (culling, skinned model transformation, ...)
+ - Porting newer game engines to the Web, and enable engines to expose more advanced rendering features. For example Unity's WebGL export uses the "mobile" feature set, but WebGPU could use a higher feature set.
+ - Porting new classes of applications to the Web: many productivity applications offload computations to the GPU and need WebGPU's support for general computations.
+ - Improving existing Web teleconferencing applications. For example, Google Meet uses machine learning to separate the user from the background. Running the machine learning in WebGPU would make it faster and more power-efficient, allowing (1) these capabilities to reach cheaper, more accessible user devices and (2) more complex and robust models.
+
+## Goals ## {#goals}
+
+Goals:
+
+ - Enable rendering of modern graphics both onscreen and offscreen.
+ - Enable general purpose computations to be executed efficiently on the GPU.
+ - Support implementation targeting various native GPU APIs: Microsoft's D3D12, Apple's Metal and Khronos' Vulkan.
+ - Provide a human-authorable language to specify computations to run on the GPU.
+ - Be implementable in the multi-process architecture of browsers and uphold the security of the Web.
+ - As much as possible, have applications work portably across different user systems and browsers.
+ - Interact with the rest of the Web platform in useful but carefully-scoped ways (essentially sharing images one way or another).
+
+Non-goals:
+
+ - Expose support for non-programmable hardware like DSPs or more fixed-function machine learning hardware.
+ - Expose support for hardware without support for general-purpose computations (like older mobile phones or even older desktop GPUs).
```

Comment:
maybe reword this? "expose support for hardware without support" is a bit repetitive